core/jvm/src/main/java/zio/internal/Signal.java

```diff
@@ -0,0 +1,151 @@
+package zio.internal;
+
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.Proxy;
+import java.util.function.Consumer;
+import java.util.logging.Logger;
+
+
+/**
+ * This class is used to handle system signals in a platform-specific way. It attemps to use the
+ * sun.misc.Signal and sun.misc.SignalHandler classes if they are available. If they are not
+ * available, it will gracefully degrade to a no-op.
+ * <p>
+ * <b>IMPLEMENTATION NOTE</b>: This class uses reflection to access the sun.misc.Signal and sun.misc.SignalHandler.
+ * This is necessary because the sun.misc.Signal and sun.misc.SignalHandler classes might not be available in some
+ * JVMs.
+ * </p>
+ */
+final class Signal {
```

Comment:
Is there a reason to write this in java?

Reply:
Actually, not really. I'm not really sure why I thought we'd need Java for this. I rewrote the implementation in Scala